### ISSUE TITLE
Lapack error in TestKissGPVariationalRegression

### ISSUE THREAD

**Issue by Balandat** [cornellius-gp/gpytorch#146]
Getting this error when running on current master:

```
RuntimeError: Lapack Error syev : 14 off-diagonal elements didn't converge to zero at caffe2/aten/src/TH/generic/THTensorLapack.cpp:388
  File "/usr/local/fbcode/gcc-5-glibc-2.23/lib/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/fbcode/gcc-5-glibc-2.23/lib/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/test/examples/test_kissgp_variational_regression.py", line 92, in test_kissgp_gp_mean_abs_error
    loss = -mll(output, y_batch)
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/gpytorch/module.py", line 184, in __call__
    outputs = self.forward(*inputs, **kwargs)
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/gpytorch/mlls/variational_marginal_log_likelihood.py", line 33, in forward
    for variational_strategy in self.model.variational_strategies()
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/gpytorch/mlls/variational_marginal_log_likelihood.py", line 33, in <genexpr>
    for variational_strategy in self.model.variational_strategies()
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/gpytorch/variational/mvn_variational_strategy.py", line 28, in kl_divergence
    inv_quad_rhs=inv_quad_rhs, log_det=True
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/gpytorch/lazy/lazy_variable.py", line 497, in inv_quad_log_det
    *args
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/gpytorch/functions/_inv_quad_log_det.py", line 143, in forward
    eigenvalues, eigenvectors = lanczos_tridiag_to_diag(t_mat)
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/gpytorch/utils/lanczos.py", line 187, in lanczos_tridiag_to_diag
    return batch_symeig(t_mat)
  File "/data/users/balandat/fbsource/fbcode/buck-out/dev-nosan/gen/pytorch/gpytorch/test_gpytorch_examples#binary,link-tree/gpytorch/utils/eig.py", line 23, in batch_symeig
    evals, evecs = mat[i, j].symeig(eigenvectors=True)
```

**Comment by jacobrgardner** [comment]
I can't reproduce this using pytorch 0.4.0, everything passes for me (and for Travis).

That error usually means there are nans somewhere, so something strange is going on.

**Comment by Balandat** [comment]
Hmm so I'm on the bleeding edge of pytorch, maybe there were some changes upstream that cause this. Let's not worry about it for now if it doesn't repro on 0.4.0.

**Comment by jacobrgardner** [comment]
Fixed by #147

**Comment by Tenoke** [comment]
I'm getting a similar error now. Not sure how related

`RuntimeError: symeig_cpu: the algorithm failed to converge; 2 off-diagonal elements of an intermediate tridiagonal form did not converge to zero.`
